presto-delta/src/main/java/com/facebook/presto/delta/DeltaExpressionUtils.java

```diff
@@ -0,0 +1,255 @@
+            {
+                if (nextItem != null) {
+                    return true;
+                }
+
+                while (inputIterator.hasNext()) {
+                    AddFile nextFile = inputIterator.next();
+                    if (evaluatePartitionPredicate(partitionPredicate, partitionColumns, typeManager, nextFile)) {
+                        nextItem = nextFile;
+                        break;
+                    }
+                }
+
+                return nextItem != null;
+            }
+
+            @Override
+            public AddFile next()
+            {
+                if (!hasNext()) {
+                    throw new NoSuchElementException("there are no more files");
+                }
+                AddFile toReturn = nextItem;
+                nextItem = null;
+                return toReturn;
+            }
+
+            @Override
+            public void close()
+                    throws IOException
+            {
+                inputIterator.close();
+            }
+        };
+    }
+
+    private static TupleDomain<String> extractPartitionColumnsPredicate(TupleDomain<DeltaColumnHandle> predicate)
+    {
+        return predicate.transform(
+                deltaColumnHandle -> {
+                    if (deltaColumnHandle.getColumnType() != PARTITION) {
+                        return null;
+                    }
+                    return deltaColumnHandle.getName();
+                });
+    }
+
+    private static boolean evaluatePartitionPredicate(
+            TupleDomain<String> partitionPredicate,
+            List<DeltaColumnHandle> partitionColumns,
+            TypeManager typeManager,
+            AddFile addFile)
+    {
+        checkArgument(!partitionPredicate.isNone(), "Expecting a predicate with at least one expression");
+        for (DeltaColumnHandle partitionColumn : partitionColumns) {
+            String columnName = partitionColumn.getName();
+            String partitionValue = addFile.getPartitionValues().get(columnName);
+            Domain domain = getDomain(partitionColumn, partitionValue, typeManager, addFile.getPath());
+            Domain columnPredicate = partitionPredicate.getDomains().get().get(columnName);
+
+            if (columnPredicate == null) {
+                continue; // there is no predicate on this column
+            }
+
+            if (columnPredicate.intersect(domain).isNone()) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private static Domain getDomain(DeltaColumnHandle columnHandle, String partitionValue, TypeManager typeManager, String filePath)
+    {
+        Type type = typeManager.getType(columnHandle.getDataType());
+        if (partitionValue == null) {
+            return Domain.onlyNull(type);
+        }
+
+        String typeBase = columnHandle.getDataType().getBase();
+        try {
+            switch (typeBase) {
+                case StandardTypes.TINYINT:
```

Comment:
static import TINYINT, SMALLINT, INTEGER, etc?

Reply:
These are used only here. I could add, but the imports become too verbose as there are too many types.